zookeeper/core/component.py

```diff
@@ -277,6 +277,64 @@ def wrapped_fn(instance) -> List[str]:
     component_cls.__dir__ = wrapped_fn
 
 
+def _wrap_configure(component_cls: Type) -> None:
+    if hasattr(component_cls, "__configure__"):
+        if not callable(component_cls.__configure__):
+            raise TypeError(
+                "The `__configure__` attribute of a @component class must be a "
+                "method."
+            )
+        call_args = inspect.signature(component_cls.__configure__).parameters
+        configure_args = inspect.signature(configure).parameters
+        configure_args = {
+            "self": configure_args["instance"],
+            **{k: v for k, v in configure_args.items() if not k == "instance"},
+        }
```

Comment:
Since we only use the names, rather than the parameters, I think this can be a set of strings:

```suggestion
        configure_args = set(
            name if name != "instance" else "self" for name in inspect.signature(configure).parameters
        )
```